.github/workflows/all-static-checks.yaml

```diff
@@ -8,6 +8,30 @@ on:
       - "main"
 
 jobs:
+  pre-commit:
+    name: Run Pre-commit Hooks and Propose Fixes
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Fetch all history so 'origin/main' is available
+          ref: ${{ github.head_ref }}
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.11
+
+      - name: Run Pre-commit and Fix Issues
+        uses: pre-commit/action@v3.0.1
+        with:
+          extra_args: "--from-ref origin/main --to-ref HEAD"
```

Comment:
Hardcoding this to head is fine, but would `${{ event.pull_request.base.ref }}` be nicer?

Reply:
```suggestion
          extra_args: "--from-ref ${{ event.pull_request.base.ref }} --to-ref HEAD"
```